instrumentation/runner/src/main/kotlin/de/mannodermaus/junit5/internal/extensions/JupiterTestMethodFinder.kt

```diff
@@ -0,0 +1,124 @@
+package de.mannodermaus.junit5.internal.extensions
+
+import android.util.Log
+import androidx.annotation.RequiresApi
+import de.mannodermaus.junit5.internal.LOG_TAG
+import org.junit.jupiter.api.RepeatedTest
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestFactory
+import org.junit.jupiter.api.TestTemplate
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.platform.commons.support.AnnotationSupport
+import org.junit.platform.commons.support.HierarchyTraversalMode
+import org.junit.platform.commons.support.ReflectionSupport
+import java.lang.reflect.Method
+import java.lang.reflect.Modifier
+import java.util.stream.Collectors
+
+internal abstract class JupiterTestMethodFinder {
+    abstract fun find(cls: Class<*>): Set<Method>
+    abstract fun hasJupiterAnnotation(method: Method): Boolean
+
+    protected val jupiterTestAnnotations = listOf(
+        Test::class.java,
+        TestFactory::class.java,
+        RepeatedTest::class.java,
+        TestTemplate::class.java,
+        ParameterizedTest::class.java,
+    )
```

Comment:
Instead of this hardcoded list, should the finder look for symbols annotated with `@Testable` (or another annotation that is itself annotated `@Testable` instead? See the Jupiter docs:

> In order to facilitate test discovery within IDEs and tools prior to launching the JUnit Platform, `TestEngine` implementations are encouraged to make use of the `@Testable` annotation. For example, the `@Test` and `@TestFactory` annotations in JUnit Jupiter are meta-annotated with @Testable. Consult the Javadoc for [@Testable](https://junit.org/junit5/docs/current/api/org.junit.platform.commons/org/junit/platform/commons/annotation/Testable.html) for further details.
>
> https://junit.org/junit5/docs/current/user-guide/#test-engines-custom

Reply:
The idea behind this declaration is to allow introspection of Jupiter's annotations by name, even for devices that don't support Java 8 yet. If we look inside of these annotations too much, we risk crashing because of unsupported features (e.g. when detecting `@Repeatable` in the meta-annotations). I do like the idea of opening up the scope beyond just Jupiter for devices that _do_ support JUnit 5, though, so I cleaned it up a little in the latest commits. ✨

That being said, most of the engines I was able to test require APIs that aren't available on Android (Spek, Cucumber). But hey: At least our Android bridge tried to load them in the first place - I call that a first step win. 🤷‍♂️

Reply:
Thanks for the explanation! That all makes sense to me!